File: c/printreverse/printreverse.c
# include <stdio.h>
# include <string.h>

void printreverse(const char* pcString);

int main(int argc, char** argv){
    const char* s = "Hello World";
    printreverse(s);
    printf("\n");
    return 0;
};

void printreverse(const char* pcString){
    if(!*pcString)
        return;
    else
        printreverse(pcString + 1);
        putchar(*pcString);
}

File: python/printreverse/printreverse.py
from sqlalchemy import null


def print_reserse(str):
    if len(str) == 0:
        return
    else:
        print_reserse(str[1:])
        print(str[0], end='')

if __name__ == '__main__':
    s = "Hello World!"
    print_reserse(s)
    print('\n')